jolt-core/src/host/mod.rs

```diff
@@ -121,11 +122,24 @@ impl Program {
         let circuit_flag_trace = trace
             .par_iter()
             .flat_map(|row| {
-                row.instruction
+                let mut bitvector = row.instruction
                     .to_circuit_flags()
                     .iter()
                     .map(|&flag| flag.into())
-                    .collect::<Vec<F>>()
+                    .collect::<Vec<F>>();
+
+                let N_FLAGS = bitvector.len();
+                bitvector.resize(N_FLAGS + InstructionSet::COUNT, F::zero());
+
+                // instruction flag
+                if let Ok(jolt_instruction) = InstructionSet::try_from(&row.instruction) {
+                    let instruction_index = InstructionSet::enum_index(&jolt_instruction);
+                    // bitvector <<= instruction_index;
+                    // bitvector |= 1;
+                    // bitvector <<= InstructionSet::COUNT - instruction_index - 1;
+                    bitvector[N_FLAGS + instruction_index] = F::from(1 as u64);
+                } 
+                bitvector
```

Comment:
We already have the instruction flag polynomials from `instruction_lookups`, shouldn't we be feeding those to the r1cs instead of recreating them here?

Reply:
Yeah this isn't ideal. I don't think the flags (either circuit or instr) should be in this `trace` function. It would be ideal to just use the opcode and get all the circuit and instruction flags in `mod.rs`. I tried doing something like this but I wasn't able to call `bitflags()` in bytecode.rs because it is expecting an ELFInstruction while we only have access to an RV32I enum in the `prove_r1cs` function. 

I might just be missing something, though. We could try to do a full refactor (no flags in trace) or a partial one for now and do the full later. What do you think?

Reply:
no what I mean is, we still need to compute `row.instruction.to_circuit_flags()` here in `trace`. But we don't need these instruction flags in the bitvector, because we already have them in the form of `InstructionPolynomials.instruction_flag_polys`
We should be passing those `instruction_flag_polys` into `prove_r1cs`, and using those in the circuit. As it is right now, we're recommitting to these instruction flag bits

Reply:
Ah right. Yeah that should be the way to do it. Lemme try adding that now.